[WIP] Amazon HealthLake Schema for FHIR Datastore resource and data-source

scottschreckengaust opened the pull request:
Just the outline for the resource and data source for Amazon HealthLake FHIR Datastore

Commit:
include additional kinesis-settings

added the following settings:
IncludeTransactionDetails
IncludePartitionValue
PartitionIncludeSchemaTable
IncludeTableAlterOperations
IncludeControlDetails
IncludeNullAndEmpty

Commit:
formatting

Commit:
added schema.Resource

added the new elements to the schema of the Kinesis settings

Commit:
fix typo

removed a "}," that should not have been removed :-)

Commit:
added TestCheckResourceAttr for Kinesis endpoints

Commit:
update fales to "false"

Commit:
update documentation

ksagle77 committed:
add changelog

ksagle77 committed:
generalize change log

drewmullen commented on aws/data_source_aws_healthlake_fhirdatastore.go:
```diff
@@ -0,0 +1,117 @@
+			},
+			"datastore_arn": {
+				Type:     schema.TypeString,
+				Required: true, // TODO: if API says "Yes", then set `Required: true`?
```
can remove `required`. schema attributes only mandate 1 callout:
- computed
- required
- optional

drewmullen commented on aws/data_source_aws_healthlake_fhirdatastore.go:
```diff
@@ -0,0 +1,117 @@
+				Type:     schema.TypeString,
+				Required: true, // API says "Yes"
+				Computed: true,
+				// ExactlyOneOf: healthlake.DatastoreStatus_Values(),
```
ValidateFunc is appropriate here instead of `exactlyoneof`

drewmullen replied:
same with L58

drewmullen commented on aws/data_source_aws_healthlake_fhirdatastore.go:
```diff
@@ -0,0 +1,117 @@
+				Default:      healthlake.FHIRVersionR4,        // "R4" if there is only one value, why not default it? Can use the SDK value too?
+			},
+			"preload_data_config": {
+				Type:     schema.TypeSet,
```
this may work but may also have to be `TypeList` - not sure

gdavison committed:
Disables linting

gdavison committed:
Moves standard AMI acctest helpers to `acctest` package

gdavison committed:
More updates to documentation

gdavison committed:
Restores importlint

gdavison committed:
Restores `go test` GitHub workflow

gdavison committed:
Restores `golangci-lint` for passing checks

gdavison committed:
Adds `nolintlint` linter

gdavison committed:
Enables `deadcode` linter

gdavison committed:
Enables `varcheck` linter

gdavison committed:
Enables `unparam` linter

gdavison committed:
Adds Semgrep rule to prevent unneeded type conversion of literal numbers

gdavison committed:
Enables `gomnd` linter. Uses forked version of `gomnd` which supports functions without a package identifier

gdavison committed:
Fixes `docs-lint`, `website-lint`, and `website-link-check` errors

gdavison committed:
Renames `ConfigLatestAmazonLinuxHVMEBSAMI()` to `ConfigLatestAmazonLinuxHvmEbsAmi()` for legibility

gdavison committed:
Restores  `terraform_providers_schema` GitHub action

gdavison committed:
Restores `tfproviderdocs` GitHub action

bambro committed:
Allow run_configurations to be set for running applications

YakDriver committed:
efs: Fix more import problems

YakDriver committed:
ecrpublic: Fix more import problems

YakDriver committed:
elasticsearch: Fix more import problems

YakDriver committed:
eks: Fix more import problems

YakDriver committed:
ecs: Fix more import problems

YakDriver committed:
elasticbeanstalk: Fix more import problems

YakDriver committed:
elasticache: Fix more import problems

YakDriver committed:
elb: Fix more import problems

YakDriver committed:
elbv2: Fix more import problems

YakDriver committed:
emr: Fix more import problems

YakDriver committed:
firehose: Fix more import problems

YakDriver committed:
glue: Fix more import problems

YakDriver committed:
glacier: Fix more import problems

YakDriver committed:
fms: Fix more import problems

YakDriver committed:
fsx: Fix more import problems

YakDriver committed:
gamelift: Fix more import problems

YakDriver committed:
globalaccelerator: Fix more import problems

YakDriver committed:
iam: Fix more import problems

YakDriver committed:
imagebuilder: Fix more import problems

YakDriver committed:
greengrass: Fix more import problems

YakDriver committed:
guardduty: Fix more import problems

YakDriver committed:
iot: Fix more import problems

YakDriver committed:
inspector: Fix more import problems

YakDriver committed:
iotevents: Fix more import problems

YakDriver committed:
iotanalytics: Fix more import problems

YakDriver committed:
kinesisvideo: Fix more import problems

YakDriver committed:
kafka: Fix more import problems

YakDriver committed:
kinesisanalytics: Fix more import problems

YakDriver committed:
kinesis: Fix more import problems

YakDriver committed:
kinesisanalyticsv2: Fix more import problems

YakDriver committed:
licensemanager: Fix more import problems

YakDriver committed:
lexmodelbuilding: Fix more import problems

YakDriver committed:
kms: Fix more import problems

YakDriver committed:
lambda: Fix more import problems

YakDriver committed:
lightsail: Fix more import problems

YakDriver committed:
macie2: Fix more import problems

YakDriver committed:
mediapackage: Fix more import problems

YakDriver committed:
mediaconvert: Fix more import problems

YakDriver committed:
mediastore: Fix more import problems

YakDriver committed:
medialive: Fix more import problems

YakDriver committed:
mediaconnect: Fix more import problems

YakDriver committed:
networkmanager: Fix more import problems

YakDriver committed:
mwaa: Fix more import problems

YakDriver committed:
mq: Fix more import problems

YakDriver committed:
networkfirewall: Fix more import problems

YakDriver committed:
neptune: Fix more import problems

YakDriver committed:
pinpoint: Fix more import problems

YakDriver committed:
organizations: Fix more import problems

YakDriver committed:
opsworks: Fix more import problems

YakDriver committed:
rds: Fix more import problems

YakDriver committed:
quicksight: Fix more import problems

YakDriver committed:
ram: Fix more import problems

YakDriver committed:
qldb: Fix more import problems

YakDriver committed:
route53: Fix more import problems

YakDriver committed:
route53recoveryreadiness: Fix more import problems

YakDriver committed:
resourcegroupstagging: Fix more import problems

YakDriver committed:
redshift: Fix more import problems

YakDriver committed:
resourcegroups: Fix more import problems

YakDriver committed:
sagemaker: Fix more import problems

YakDriver committed:
route53resolver: Fix more import problems

YakDriver committed:
s3control: Fix more import problems

YakDriver committed:
s3: Fix more import problems

YakDriver committed:
schemas: Fix more import problems

YakDriver committed:
ses: Fix more import problems

YakDriver committed:
sfn: Fix more import problems

YakDriver committed:
servicecatalog: Fix more import problems

YakDriver committed:
serverlessapprepo: Fix more import problems

YakDriver committed:
servicediscovery: Fix more import problems

YakDriver committed:
securityhub: Fix more import problems

YakDriver committed:
secretsmanager: Fix more import problems

YakDriver committed:
signer: Fix more import problems

YakDriver committed:
ssm: Fix more import problems

YakDriver committed:
sqs: Fix more import problems

YakDriver committed:
sns: Fix more import problems

YakDriver committed:
shield: Fix more import problems

YakDriver committed:
storagegateway: Fix more import problems

YakDriver committed:
synthetics: Fix more import problems

YakDriver committed:
ssoadmin: Fix more import problems

YakDriver committed:
timestreamwrite: Fix more import problems

YakDriver committed:
swf: Fix more import problems

YakDriver committed:
transfer: Fix more import problems

YakDriver committed:
wafregional: Fix more import problems

YakDriver committed:
waf: Fix more import problems

YakDriver committed:
worklink: Fix more import problems

YakDriver committed:
wafv2: Fix more import problems

YakDriver committed:
workspaces: Fix more import problems

YakDriver committed:
xray: Fix more import problems

YakDriver committed:
After import_4

YakDriver committed:
After namefix

YakDriver committed:
ssoadmin: Fix more import problems

YakDriver committed:
After import_5

YakDriver committed:
After fmt_3

Commit:
v3.63.0

YakDriver committed:
Point golangci to right place

Commit:
Update CHANGELOG.md after v3.63.0

YakDriver committed:
Merge pull request #21306 from hashicorp/f-yakdriver-refactor-3

Refactor to Service Packages

gdavison committed:
Restores code generation checks

YakDriver committed:
Remove unnecessary artifact of 21306

YakDriver committed:
Merge pull request #21312 from hashicorp/td-remove-uuid-placeholder

Remove unnecessary artifact of 21306

Commit:
add additional not found error handling for dynamodb and ecs tag resources

YakDriver committed:
tests: Clean up names

YakDriver committed:
tests: Clean up names

YakDriver committed:
Clean up names

YakDriver committed:
Merge pull request #21321 from hashicorp/td-kinesisanalyticsv2-sp-updates

kinesisanalyticsv2: Clean up function names

Commit:
CR update; pass in parent not found related content as arguments

anGie44 committed:
Merge pull request #21256 from hashicorp/f-notfound-error-tags

add "parent not found error" handling

gdavison committed:
Merge pull request #21186 from hashicorp/svcpkg-docs

docs: Updates contributing docs for Service Packages

gdavison committed:
Extends `golangci-lint` timeout to 15 minutes

YakDriver committed:
Export funcs for region data source

YakDriver committed:
Update for region data source move to NAS

YakDriver committed:
Merge pull request #21325 from hashicorp/sp-move-regions-to-nas

Move region/s data sources to NAS

gdavison committed:
Merge pull request #21266 from hashicorp/svcpkg-linting

linting: Updates linting for Service Packages

YakDriver committed:
Update labels, add new, alphabetize

YakDriver committed:
Update labeler PR triage for SP

YakDriver committed:
Merge pull request #21326 from hashicorp/repo-labelers

Update labels, labeler for service packages

zhelding committed:
new file: docs/contributing/service-package-pullrequest-guide.md

zhelding committed:
run make docs-lint-fix

zhelding committed:
amend docs header for consistency

gdavison committed:
Updates generator documentation

gdavison committed:
No longer exports listpages functions

Commit:
adjust directives and add logic where needed to templates

Commit:
Merge branch 'main' into sp-make-gen

kamilturek committed:
Add new data source: aws_iam_user_ssh_key

kamilturek committed:
Create documentation file for data source: aws_iam_user_ssh_key

kamilturek committed:
Add changelog entry

kamilturek committed:
Fix typo

PatMyron committed:
provider: cn-northwest-1 (Ningxia) hostedZoneID

https://docs.aws.amazon.com/general/latest/gr/s3.html#s3_region

kamilturek committed:
Include source type in request if event categories have changed

kamilturek committed:
Update acceptance tests to avoid regression

kamilturek committed:
Add changelog

ewbankkit committed:
r/aws_dms_endpoint: Additional Kinesis settings.

Acceptance test output:

$ make testacc TESTARGS='-run=TestAccDMSEndpoint_kinesis' PKG_NAME=internal/service/dms
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/dms -v -count 1 -parallel 20 -run=TestAccDMSEndpoint_kinesis -timeout 180m
=== RUN   TestAccDMSEndpoint_kinesis
=== PAUSE TestAccDMSEndpoint_kinesis
=== CONT  TestAccDMSEndpoint_kinesis
--- PASS: TestAccDMSEndpoint_kinesis (182.20s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/dms	182.255s

ewbankkit committed:
Update CHANGELOG entry.

bambro committed:
refactor: apply suggestions from code review

Co-authored-by: Dirk Avery <31492422+YakDriver@users.noreply.github.com>

bambro committed:
fix: update additional test method references

dependabot[bot] committed:
build(deps): bump integrations/github in /infrastructure/repository

Bumps [integrations/github](https://github.com/integrations/terraform-provider-github) from 4.16.0 to 4.17.0.
- [Release notes](https://github.com/integrations/terraform-provider-github/releases)
- [Changelog](https://github.com/integrations/terraform-provider-github/blob/main/CHANGELOG.md)
- [Commits](https://github.com/integrations/terraform-provider-github/compare/v4.16.0...v4.17.0)

---
updated-dependencies:
- dependency-name: integrations/github
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump github.com/aws/aws-sdk-go in /providerlint

Bumps [github.com/aws/aws-sdk-go](https://github.com/aws/aws-sdk-go) from 1.41.2 to 1.41.4.
- [Release notes](https://github.com/aws/aws-sdk-go/releases)
- [Changelog](https://github.com/aws/aws-sdk-go/blob/main/CHANGELOG.md)
- [Commits](https://github.com/aws/aws-sdk-go/compare/v1.41.2...v1.41.4)

---
updated-dependencies:
- dependency-name: github.com/aws/aws-sdk-go
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

bambro committed:
Merge branch 'main' into f-kinesis-application-run-configurations

bambro committed:
refactor: fix up latest merge

ewbankkit committed:
Simplify comments.

Commit:
mardown lint

Commit:
add comments for new bool template data; add usage for untag bool flag

YakDriver committed:
Merge pull request #21343 from hashicorp/dependabot/go_modules/providerlint/github.com/aws/aws-sdk-go-1.41.4

build(deps): bump github.com/aws/aws-sdk-go from 1.41.2 to 1.41.4 in /providerlint

YakDriver committed:
Merge pull request #21342 from hashicorp/dependabot/terraform/infrastructure/repository/integrations/github-4.17.0

build(deps): bump integrations/github from 4.16.0 to 4.17.0 in /infrastructure/repository

anGie44 committed:
Merge pull request #21333 from hashicorp/sp-make-gen

fix: adjust generators directives with bool flags and add conditional logic for header template

YakDriver committed:
Merge pull request #21303 from bambro/f-kinesis-application-run-configurations

fix: Allow run_configurations to be set on Kinesis AnalyticsV2

gdavison committed:
Restores `providerlint` linting for passing cases and simple fixes

gdavison committed:
Excludes renamed data source test files

ewbankkit committed:
Merge pull request #21337 from PatMyron/patch-1

provider: cn-northwest-1 (Ningxia) hostedZoneID

ewbankkit committed:
Merge pull request #21338 from kamilturek/b-aws_db_event_subscription_update

r/db_event_subscription - fix adding new `event_categories` to existing resource

ewbankkit committed:
Merge pull request #20084 from ksagle77/f-aws_dms_endpoint-add-setting

Add support for additional kinesis_settings to dms_endpoint

gdavison committed:
Restores `AWSAT003` check

ewbankkit committed:
Missing CHANGELOG entries.

ewbankkit committed:
Merge pull request #21352 from hashicorp/td-various-changelog-entries

Missing CHANGELOG entries

Commit:
Update CHANGELOG.md for #21352

gdavison committed:
Restores `AWSAT005` check

gdavison committed:
Restores `providerlint` in Makefile

gdavison committed:
Adds output lines to linters

Commit:
move test fixture from apigateway to apigatewayv2, where it is used

gdavison committed:
Merge pull request #21330 from hashicorp/svcpkg-linting-deadcode

linting: Restores unused code linting for Service Packages

gdavison committed:
Extends `golangci-lint` timeout to 20 minutes

zhelding committed:
Remove trailing space

zhelding committed:
Add guide step for provider.go

gdavison committed:
Restores `terrafmt` for acceptance tests

Commit:
Merge branch 'main' into t-apigatewayv2-missing-fixture

Commit:
update configuration's listener protocol type

gdavison committed:
Restores `validate-terraform` for acceptance tests

Commit:
fix: pass in correct deployment group name when unchanged

nyamada100 committed:
fix and test #8854

Commit:
Update CHANGELOG for #21362

Commit:
get tags with PropagateAtLaunch key instead of propagate_at_launch

nyamada100 committed:
add changelog

dependabot[bot] committed:
build(deps): bump github.com/aws/aws-sdk-go in /providerlint

Bumps [github.com/aws/aws-sdk-go](https://github.com/aws/aws-sdk-go) from 1.41.4 to 1.41.5.
- [Release notes](https://github.com/aws/aws-sdk-go/releases)
- [Changelog](https://github.com/aws/aws-sdk-go/blob/main/CHANGELOG.md)
- [Commits](https://github.com/aws/aws-sdk-go/compare/v1.41.4...v1.41.5)

---
updated-dependencies:
- dependency-name: github.com/aws/aws-sdk-go
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

ewbankkit committed:
Add 'internal/service/cloudwatch/enum.go'.

zhelding committed:
Update mv instruction to git mv

anGie44 committed:
Merge pull request #21364 from hashicorp/b-asg-tag-gen

generator/tags: include `PropagateAtLaunch` key in TagData for `autoscaling`

anGie44 committed:
Merge pull request #21362 from hashicorp/b-codedeploy-deployment-group

r/codedeploy/deployment_group: fix `deployment_group_name` update handling

anGie44 committed:
Merge pull request #21360 from hashicorp/t-appmesh-vgateway-listener-health-chcks

tests/appmesh_virtual_gateway: update configuration's listener protocol type

anGie44 committed:
Merge pull request #21354 from hashicorp/t-apigatewayv2-missing-fixture

tests/apigatewayv2: move test fixture from apigateway to apigatewayv2

Commit:
deactivate keys with 'ACTION_NEEDED' status as well

Commit:
Update CHANGELOG.md for #21354

Commit:
Update CHANGELOG for #21369

ewbankkit committed:
Prefer Redshift `dc2` node types over `dc1`. cloudwatchevents.

Acceptance test output:

% make testacc TESTARGS='-run=TestAccCloudWatchEventsTarget_redshift' PKG_NAME=internal/service/cloudwatchevents
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/cloudwatchevents -v -count 1 -parallel 20 -run=TestAccCloudWatchEventsTarget_redshift -timeout 180m
=== RUN   TestAccCloudWatchEventsTarget_redshift
=== PAUSE TestAccCloudWatchEventsTarget_redshift
=== CONT  TestAccCloudWatchEventsTarget_redshift
--- PASS: TestAccCloudWatchEventsTarget_redshift (220.76s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/cloudwatchevents	223.878s

ewbankkit committed:
Merge pull request #21363 from nyamada00/b-aws_metric_alarm-8854

default value 'missing' for 'treat_missing_data'

ewbankkit committed:
Prefer Redshift `dc2` node types over `dc1`. firehose.

Acceptance test output:

% make testacc TESTARGS='-run=TestAccFirehoseDeliveryStream_redshift' PKG_NAME=internal/service/firehose
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/firehose -v -count 1 -parallel 20 -run=TestAccFirehoseDeliveryStream_redshift -timeout 180m
=== RUN   TestAccFirehoseDeliveryStream_redshiftUpdates
=== PAUSE TestAccFirehoseDeliveryStream_redshiftUpdates
=== CONT  TestAccFirehoseDeliveryStream_redshiftUpdates
--- PASS: TestAccFirehoseDeliveryStream_redshiftUpdates (373.15s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/firehose	378.025s

Commit:
set accesslevelfilter value with 'self' in provisioned product sweeper

zhelding committed:
Merge pull request #21357 from hashicorp/d-service-package-pr-guide

Add service package refactor PR fix guide

Commit:
Update CHANGELOG.md for #21357

ewbankkit committed:
Prefer Redshift `dc2` node types over `dc1`. aws_redshift_scheduled_action.

Acceptance test output:

% make testacc TESTARGS='-run=TestAccRedshiftScheduledAction_resizeClusterWithOptions' PKG_NAME=internal/service/redshift
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/redshift -v -count 1 -parallel 20 -run=TestAccRedshiftScheduledAction_resizeClusterWithOptions -timeout 180m
=== RUN   TestAccRedshiftScheduledAction_resizeClusterWithOptions
=== PAUSE TestAccRedshiftScheduledAction_resizeClusterWithOptions
=== CONT  TestAccRedshiftScheduledAction_resizeClusterWithOptions
--- PASS: TestAccRedshiftScheduledAction_resizeClusterWithOptions (30.36s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/redshift	35.338s

Commit:
update sweep target to include predetermined sweep dir and tags flag

ewbankkit committed:
Noun form is 'refactoring'.

ewbankkit committed:
Add info on moving finder and status functions.

anGie44 committed:
Merge pull request #21369 from hashicorp/td-route53-key-signing-key-delete

r/route53_key_signing_key: also deactivate keys with `ACTION_NEEDED` status

anGie44 committed:
Merge pull request #21372 from hashicorp/makefile-sweep-target

GNUmakefile: update `sweep` target to include predetermined sweep dir and tags flag

Commit:
Update CHANGELOG.md for #21372

ewbankkit committed:
Prefer Redshift `dc2` node types over `dc1`. d/aws_redshift_cluster.

Acceptance test output:

% make testacc TESTARGS='-run=TestAccRedshiftClusterDataSource_' PKG_NAME=internal/service/redshift
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/redshift -v -count 1 -parallel 20 -run=TestAccRedshiftClusterDataSource_ -timeout 180m
=== RUN   TestAccRedshiftClusterDataSource_basic
=== PAUSE TestAccRedshiftClusterDataSource_basic
=== RUN   TestAccRedshiftClusterDataSource_vpc
=== PAUSE TestAccRedshiftClusterDataSource_vpc
=== RUN   TestAccRedshiftClusterDataSource_logging
=== PAUSE TestAccRedshiftClusterDataSource_logging
=== CONT  TestAccRedshiftClusterDataSource_basic
=== CONT  TestAccRedshiftClusterDataSource_logging
=== CONT  TestAccRedshiftClusterDataSource_vpc
--- PASS: TestAccRedshiftClusterDataSource_vpc (227.38s)
--- PASS: TestAccRedshiftClusterDataSource_logging (259.68s)
--- PASS: TestAccRedshiftClusterDataSource_basic (273.12s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/redshift	276.206s

ewbankkit committed:
Merge pull request #21373 from hashicorp/doc-service-package-pullrequest-guide-additions

Updates to "Service Package Refactor Pull Request Guide"

anGie44 committed:
Merge pull request #21371 from hashicorp/td-servicecat-provisioned-prod-sweep

sweep/servicatalog_provisioned_product: set `accessLevelFilter.value` with `self`

ewbankkit committed:
Prefer Redshift `dc2` node types over `dc1`. r/aws_redshift_cluster.

Acceptance test output:

% make testacc TESTARGS='-run=TestAccRedshiftCluster_\|TestAccRedshiftSnapshotSchedule_\|TestAccRedshiftSnapshotScheduleAssociation_' PKG_NAME=internal/service/redshift
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/redshift -v -count 1 -parallel 20 -run=TestAccRedshiftCluster_\|TestAccRedshiftSnapshotSchedule_\|TestAccRedshiftSnapshotScheduleAssociation_ -timeout 180m
=== RUN   TestAccRedshiftCluster_basic
=== PAUSE TestAccRedshiftCluster_basic
=== RUN   TestAccRedshiftCluster_disappears
=== PAUSE TestAccRedshiftCluster_disappears
=== RUN   TestAccRedshiftCluster_withFinalSnapshot
=== PAUSE TestAccRedshiftCluster_withFinalSnapshot
=== RUN   TestAccRedshiftCluster_kmsKey
=== PAUSE TestAccRedshiftCluster_kmsKey
=== RUN   TestAccRedshiftCluster_enhancedVPCRoutingEnabled
=== PAUSE TestAccRedshiftCluster_enhancedVPCRoutingEnabled
=== RUN   TestAccRedshiftCluster_loggingEnabled
=== PAUSE TestAccRedshiftCluster_loggingEnabled
=== RUN   TestAccRedshiftCluster_snapshotCopy
=== PAUSE TestAccRedshiftCluster_snapshotCopy
=== RUN   TestAccRedshiftCluster_iamRoles
=== PAUSE TestAccRedshiftCluster_iamRoles
=== RUN   TestAccRedshiftCluster_publiclyAccessible
=== PAUSE TestAccRedshiftCluster_publiclyAccessible
=== RUN   TestAccRedshiftCluster_updateNodeCount
=== PAUSE TestAccRedshiftCluster_updateNodeCount
=== RUN   TestAccRedshiftCluster_updateNodeType
=== PAUSE TestAccRedshiftCluster_updateNodeType
=== RUN   TestAccRedshiftCluster_tags
=== PAUSE TestAccRedshiftCluster_tags
=== RUN   TestAccRedshiftCluster_forceNewUsername
=== PAUSE TestAccRedshiftCluster_forceNewUsername
=== RUN   TestAccRedshiftCluster_changeAvailabilityZone
=== PAUSE TestAccRedshiftCluster_changeAvailabilityZone
=== RUN   TestAccRedshiftCluster_changeEncryption1
=== PAUSE TestAccRedshiftCluster_changeEncryption1
=== RUN   TestAccRedshiftCluster_changeEncryption2
=== PAUSE TestAccRedshiftCluster_changeEncryption2
=== RUN   TestAccRedshiftSnapshotScheduleAssociation_basic
=== PAUSE TestAccRedshiftSnapshotScheduleAssociation_basic
=== RUN   TestAccRedshiftSnapshotSchedule_basic
=== PAUSE TestAccRedshiftSnapshotSchedule_basic
=== RUN   TestAccRedshiftSnapshotSchedule_withMultipleDefinition
=== PAUSE TestAccRedshiftSnapshotSchedule_withMultipleDefinition
=== RUN   TestAccRedshiftSnapshotSchedule_withIdentifierPrefix
=== PAUSE TestAccRedshiftSnapshotSchedule_withIdentifierPrefix
=== RUN   TestAccRedshiftSnapshotSchedule_withDescription
=== PAUSE TestAccRedshiftSnapshotSchedule_withDescription
=== RUN   TestAccRedshiftSnapshotSchedule_withTags
=== PAUSE TestAccRedshiftSnapshotSchedule_withTags
=== RUN   TestAccRedshiftSnapshotSchedule_withForceDestroy
=== PAUSE TestAccRedshiftSnapshotSchedule_withForceDestroy
=== CONT  TestAccRedshiftCluster_basic
=== CONT  TestAccRedshiftCluster_forceNewUsername
=== CONT  TestAccRedshiftSnapshotSchedule_withForceDestroy
=== CONT  TestAccRedshiftCluster_snapshotCopy
=== CONT  TestAccRedshiftCluster_tags
=== CONT  TestAccRedshiftCluster_updateNodeType
=== CONT  TestAccRedshiftCluster_updateNodeCount
=== CONT  TestAccRedshiftCluster_publiclyAccessible
=== CONT  TestAccRedshiftCluster_iamRoles
=== CONT  TestAccRedshiftCluster_changeEncryption2
=== CONT  TestAccRedshiftSnapshotSchedule_basic
=== CONT  TestAccRedshiftSnapshotSchedule_withMultipleDefinition
=== CONT  TestAccRedshiftSnapshotScheduleAssociation_basic
=== CONT  TestAccRedshiftCluster_kmsKey
=== CONT  TestAccRedshiftCluster_loggingEnabled
=== CONT  TestAccRedshiftCluster_changeEncryption1
=== CONT  TestAccRedshiftCluster_enhancedVPCRoutingEnabled
=== CONT  TestAccRedshiftCluster_changeAvailabilityZone
=== CONT  TestAccRedshiftSnapshotSchedule_withIdentifierPrefix
=== CONT  TestAccRedshiftSnapshotSchedule_withTags
=== CONT  TestAccRedshiftCluster_withFinalSnapshot
--- PASS: TestAccRedshiftSnapshotSchedule_withIdentifierPrefix (29.41s)
=== CONT  TestAccRedshiftCluster_disappears
--- PASS: TestAccRedshiftSnapshotSchedule_basic (39.86s)
=== CONT  TestAccRedshiftSnapshotSchedule_withDescription
--- PASS: TestAccRedshiftSnapshotSchedule_withMultipleDefinition (41.37s)
--- PASS: TestAccRedshiftSnapshotSchedule_withTags (41.43s)
--- PASS: TestAccRedshiftSnapshotSchedule_withDescription (12.74s)
--- PASS: TestAccRedshiftCluster_snapshotCopy (236.79s)
--- PASS: TestAccRedshiftCluster_tags (237.85s)
--- PASS: TestAccRedshiftSnapshotSchedule_withForceDestroy (276.67s)
--- PASS: TestAccRedshiftSnapshotScheduleAssociation_basic (284.01s)
--- PASS: TestAccRedshiftCluster_iamRoles (357.76s)
--- PASS: TestAccRedshiftCluster_enhancedVPCRoutingEnabled (371.66s)
--- PASS: TestAccRedshiftCluster_basic (461.33s)
--- PASS: TestAccRedshiftCluster_changeAvailabilityZone (481.58s)
--- PASS: TestAccRedshiftCluster_kmsKey (541.35s)
--- PASS: TestAccRedshiftCluster_loggingEnabled (577.85s)
--- PASS: TestAccRedshiftCluster_disappears (567.11s)
--- PASS: TestAccRedshiftCluster_publiclyAccessible (624.01s)
--- PASS: TestAccRedshiftCluster_withFinalSnapshot (793.65s)
--- PASS: TestAccRedshiftCluster_forceNewUsername (814.79s)
--- PASS: TestAccRedshiftCluster_updateNodeType (1424.68s)
--- PASS: TestAccRedshiftCluster_changeEncryption2 (1503.98s)
--- PASS: TestAccRedshiftCluster_changeEncryption1 (1561.05s)
--- PASS: TestAccRedshiftCluster_updateNodeCount (1656.23s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/redshift	1659.067s

YakDriver committed:
Update labeler for repository, pre-service-packages; cleanup

YakDriver committed:
Add pre-service-packages label

YakDriver committed:
Fix tests labeler

YakDriver committed:
Merge pull request #21381 from hashicorp/r-labels-labeler-update

repo: Labels/labeler update

YakDriver committed:
Merge pull request #21366 from hashicorp/dependabot/go_modules/providerlint/github.com/aws/aws-sdk-go-1.41.5

build(deps): bump github.com/aws/aws-sdk-go from 1.41.4 to 1.41.5 in /providerlint

dependabot[bot] committed:
build(deps): bump github.com/aws/aws-sdk-go from 1.41.2 to 1.41.6

Bumps [github.com/aws/aws-sdk-go](https://github.com/aws/aws-sdk-go) from 1.41.2 to 1.41.6.
- [Release notes](https://github.com/aws/aws-sdk-go/releases)
- [Changelog](https://github.com/aws/aws-sdk-go/blob/main/CHANGELOG.md)
- [Commits](https://github.com/aws/aws-sdk-go/compare/v1.41.2...v1.41.6)

---
updated-dependencies:
- dependency-name: github.com/aws/aws-sdk-go
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

gdavison committed:
Removes `goimports` as it is no longer needed

gdavison committed:
Merge pull request #21356 from hashicorp/svcpkg-codegen

linting: Restore code generation checks

gdavison committed:
Merge pull request #21332 from hashicorp/svcpkg-linting-gomnd

linting: Restores `gomnd` linter

suzuki-shunsuke committed:
fix: fix make testacc

Commit:
ignore 'state' value during import step of acctest

ewbankkit committed:
Merge pull request #21385 from suzuki-shunsuke/fix/fix-make-testacc

fix: fix make testacc

YakDriver committed:
Merge pull request #21382 from hashicorp/dependabot/go_modules/github.com/aws/aws-sdk-go-1.41.6

build(deps): bump github.com/aws/aws-sdk-go from 1.41.2 to 1.41.6

dependabot[bot] committed:
build(deps): bump github.com/aws/aws-sdk-go in /providerlint

Bumps [github.com/aws/aws-sdk-go](https://github.com/aws/aws-sdk-go) from 1.41.5 to 1.41.6.
- [Release notes](https://github.com/aws/aws-sdk-go/releases)
- [Changelog](https://github.com/aws/aws-sdk-go/blob/main/CHANGELOG.md)
- [Commits](https://github.com/aws/aws-sdk-go/compare/v1.41.5...v1.41.6)

---
updated-dependencies:
- dependency-name: github.com/aws/aws-sdk-go
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

anGie44 committed:
Merge pull request #21387 from hashicorp/td-appconfig-deployment-waiters

tests/appconfig_deployment: ignore `state` value during import step of `TestAccAppConfigDeployment_predefinedStrategy`

ewbankkit committed:
Merge pull request #21376 from hashicorp/t-redshift-cluster-dc2

Prefer Redshift `dc2` node types over `dc1` in acceptance test configurations

ewbankkit committed:
Merge pull request #21390 from hashicorp/dependabot/go_modules/providerlint/github.com/aws/aws-sdk-go-1.41.6

build(deps): bump github.com/aws/aws-sdk-go from 1.41.5 to 1.41.6 in /providerlint

ewbankkit committed:
Merge pull request #21335 from kamilturek/f-data_source_aws_iam_user_ssh_key

d/iam_user_ssh_key - new data source

gdavison committed:
Merge pull request #21331 from hashicorp/svcpkg-linting-unparam

linting: Restores `unparam` linter

gdavison committed:
Merge pull request #21353 from hashicorp/svgpkg-providerlint

linting: Restores `providerlint` (formerly `awsproviderlint`)

gdavison committed:
Merge pull request #21383 from hashicorp/svgpkg-acctest-tf-validation

linting: Restore Terraform code validation for acceptance tests

Commit:
Update CHANGELOG.md for #21383

scottschreckengaust committed:
Service Package Refactor Merge

scottschreckengaust committed:
Fixing schema

scottschreckengaust committed:
WIP savepoint

scottschreckengaust committed:
Adding healthlake provider
